Use correct version of hydra-node for script publishing

noonio opened the pull request:
This fixes the network tests by using the repo-version of hydra-node (i.e. the same one as the docker images that are run eventually, via the docker-compose-netem.yaml file.)

noonio committed:
Hack some env var usage in

ch1bo commented on demo/export-tx-id-and-pparams.sh:
```diff
@@ -6,15 +6,15 @@ SCRIPT_DIR=${SCRIPT_DIR:-$(realpath $(dirname $(realpath $0)))}
     echo >&2 "Using provided cardano-cli command: ${1}"
     $(${1} version > /dev/null)
     CCLI_CMD=${1}
     DEVNET_DIR=${SCRIPT_DIR}/devnet
```
Why is DEVNET_DIR not always relative to SCRIPT_DIR?

noonio replied:
Because, if you're running a docker image, it's (assumed to be) mounted directly at `/devnet`

🥲 🥲 🥲

ch1bo commented on .github/workflows/network-test.yaml:
```diff
@@ -67,7 +67,7 @@ jobs:
         # :tear: socket permissions.
         sudo chown runner:docker devnet/node.socket
-        ./export-tx-id-and-pparams.sh
+        DEVNET_DIR=devnet HYDRA_NODE_CMD="nix run .#hydra-node --" ./export-tx-id-and-pparams.sh
```
At this stage I would recommend to just put the `nix run .#hydra-node publish-scripts..` invocation directly here.

ch1bo replied:
Done this now in https://github.com/cardano-scaling/hydra/commit/67ef048ea9a4dacdb13f7a69f627e32e968f3619

noonio replied:
Seems reasonable; the only reason I didn't do that is I was wondering if the intention was to support people who choose not to run Nix at all.

ch1bo reviewed (approved):
Good for a quick fix, but we all know this asks for some refactoring.

ch1bo committed:
Moves publish/protocol param prep back into seed-devnet

The overriding of commands from the network-tests usage site came to a
level where simply calling the hydra-node publish-scripts and cli
query protocol-parameters seemed easier than maintaining this second
shell script.
